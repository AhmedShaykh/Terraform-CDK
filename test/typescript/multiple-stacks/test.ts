/**
 * Testing a full cycle of diff, deploy and destroy
 *
 * @group typescript
 */

import { TestDriver } from "../../test-helper";

describe("full integration test", () => {
  let driver: TestDriver;

  beforeAll(() => {
    driver = new TestDriver(__dirname)
    driver.setupTypescriptProject()
  });

  test("synth", () => {
    driver.synth()
    expect(driver.synthesizedStack('first')).toMatchSnapshot()
    expect(driver.synthesizedStack('second')).toMatchSnapshot()
  });

  test("synth with json output", () => {
    expect(driver.synth('--json')).toMatchSnapshot()
  });

  test("diff", () => {
    expect(driver.diff('first')).toMatchInlineSnapshot(`
      "Stack: first
      Resources
       + NULL_RESOURCE       test                null_resource.test


      Diff: 1 to create, 0 to update, 0 to delete.
      "
    `);


    expect(driver.diff('second')).toMatchInlineSnapshot(`
      "Stack: second
      Resources
       + NULL_RESOURCE       test                null_resource.test


      Diff: 1 to create, 0 to update, 0 to delete.
      "
    `);

    expect(() => driver.diff()).toThrowError('Found more than one stack')
  });

  test("list", () => {
    expect(driver.list()).toMatchInlineSnapshot(`
      "Stack name                      Path
      first                           cdktf.out/stacks/first
      second                          cdktf.out/stacks/second
      "
    `);
  });

  test("deploy", () => {
    expect(driver.deploy('first')).toMatchInlineSnapshot(`
      "Deploying Stack: first
      Resources
       ✔ NULL_RESOURCE       test                null_resource.test


      Summary: 1 created, 0 updated, 0 destroyed.
      "
    `);

    expect(driver.deploy('second')).toMatchInlineSnapshot(`
      "Deploying Stack: second
      Resources
       ✔ NULL_RESOURCE       test                null_resource.test


      Summary: 1 created, 0 updated, 0 destroyed.
      "
    `);

    let error;
    try { driver.deploy() } catch(e) { error = e.message }
    expect(error).toMatch('Found more than one stack, please specify a target stack first, second');
  });

  test("destroy", () => {
    expect(driver.destroy('first')).toMatchInlineSnapshot(`
      "Destroying Stack: first
      Resources
       ✔ NULL_RESOURCE       test                null_resource.test


      Summary: 1 destroyed.
      "
    `);

    expect(driver.destroy('second')).toMatchInlineSnapshot(`
      "Destroying Stack: second
      Resources
       ✔ NULL_RESOURCE       test                null_resource.test


      Summary: 1 destroyed.
      "
    `);

    let error;
    try { driver.destroy() } catch(e) { error = e.message }
    expect(error).toMatch('Found more than one stack, please specify a target stack first, second');
  });
});
