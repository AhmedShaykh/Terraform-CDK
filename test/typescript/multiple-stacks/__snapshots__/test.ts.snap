// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`full integration test synth 1`] = `
"{
  \\"//\\": {
    \\"metadata\\": {
      \\"version\\": \\"stubbed\\",
      \\"stackName\\": \\"first\\",
      \\"backend\\": \\"local\\",
      \\"overrides\\": {
        \\"null_resource\\": [
          \\"provisioner\\"
        ]
      }
    }
  },
  \\"terraform\\": {
    \\"required_providers\\": {
      \\"null\\": {
        \\"version\\": \\" ~> 3.1.0\\",
        \\"source\\": \\"null\\"
      }
    },
    \\"backend\\": {
      \\"local\\": {
        \\"path\\": \\"terraform.tfstate\\"
      }
    }
  },
  \\"provider\\": {
    \\"null\\": [
      {}
    ]
  },
  \\"resource\\": {
    \\"null_resource\\": {
      \\"test\\": {
        \\"provisioner\\": [
          {
            \\"local-exec\\": {
              \\"command\\": \\"echo \\\\\\"hello deploy\\\\\\"\\"
            }
          }
        ],
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"first/test\\",
            \\"uniqueId\\": \\"test\\"
          }
        }
      }
    }
  }
}"
`;

exports[`full integration test synth 2`] = `
"{
  \\"//\\": {
    \\"metadata\\": {
      \\"version\\": \\"stubbed\\",
      \\"stackName\\": \\"second\\",
      \\"backend\\": \\"local\\",
      \\"overrides\\": {
        \\"null_resource\\": [
          \\"provisioner\\"
        ]
      }
    }
  },
  \\"terraform\\": {
    \\"required_providers\\": {
      \\"null\\": {
        \\"version\\": \\" ~> 3.1.0\\",
        \\"source\\": \\"null\\"
      }
    },
    \\"backend\\": {
      \\"local\\": {
        \\"path\\": \\"terraform.tfstate\\"
      }
    }
  },
  \\"provider\\": {
    \\"null\\": [
      {}
    ]
  },
  \\"resource\\": {
    \\"null_resource\\": {
      \\"test\\": {
        \\"provisioner\\": [
          {
            \\"local-exec\\": {
              \\"command\\": \\"echo \\\\\\"hello deploy\\\\\\"\\"
            }
          }
        ],
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"second/test\\",
            \\"uniqueId\\": \\"test\\"
          }
        }
      }
    }
  }
}"
`;

exports[`full integration test synth with json output 1`] = `
"{
  \\"//\\": {
    \\"metadata\\": {
      \\"version\\": \\"stubbed\\",
      \\"stackName\\": \\"first\\",
      \\"backend\\": \\"local\\",
      \\"overrides\\": {
        \\"null_resource\\": [
          \\"provisioner\\"
        ]
      }
    }
  },
  \\"terraform\\": {
    \\"required_providers\\": {
      \\"null\\": {
        \\"version\\": \\" ~> 3.1.0\\",
        \\"source\\": \\"null\\"
      }
    },
    \\"backend\\": {
      \\"local\\": {
        \\"path\\": \\"terraform.tfstate\\"
      }
    }
  },
  \\"provider\\": {
    \\"null\\": [
      {}
    ]
  },
  \\"resource\\": {
    \\"null_resource\\": {
      \\"test\\": {
        \\"provisioner\\": [
          {
            \\"local-exec\\": {
              \\"command\\": \\"echo \\\\\\"hello deploy\\\\\\"\\"
            }
          }
        ],
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"first/test\\",
            \\"uniqueId\\": \\"test\\"
          }
        }
      }
    }
  }
}
"
`;
